### PR TITLE
EPMLSTRTAI-338: More detailed exception

### PR DESCRIPTION
**Pull request by Walrus94**

### CODE REVIEW

File: src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java [hunk 1]
```diff
@@ -40,7 +43,21 @@ public boolean isValid(SearchRule searchRule, ValidationContext validationContex
...
+                .map(url -> {
+                    try {
+                        return  new Pair<>(url, searchRule.extractElementsFromWebSite(url).size());
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                    return null;
+                })
+                .filter(pair -> pair.getValue() > 1)
+                .map(pair -> "On url: " + pair.getKey() + " founded " + pair.getValue() + " elements")
```
Comment (pomkine) [inline src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java]: founded? or found? :)

File: src/main/java/com/epam/page/object/generator/validators/ValidationResult.java [hunk 2]
```diff
@@ -5,15 +5,17 @@
 public class ValidationResult {
 
     private boolean isValid;
-    private Validator validator;
+    private AbstractValidator validator;
```
Comment (pomkine) [inline src/main/java/com/epam/page/object/generator/validators/ValidationResult.java]: use interface instead

File: src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java [hunk 3]
```diff
@@ -40,7 +43,21 @@ public boolean isValid(SearchRule searchRule, ValidationContext validationContex
...
 
     @Override
-    public String getExceptionMessage() {
-        return "This element is not uniqueness";
+    public String getExceptionMessage(SearchRule searchRule, ValidationContext validationContext) {
+       // return "This element is not uniqueness";
+        return validationContext.getUrls()
+                .stream()
+                .map(url -> {
+                    try {
```
Comment (kugushev) [inline src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java]: 7-lines lambda is a bad style. Let's extract a method

**Review by kugushev — APPROVED** [review]

File: src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java (hunk 3 above)
Comment (iteaky) [reply src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java]: @AleksandrKugushev  just extract lambda to method and use it inside method getExceptionMessage()???
or exctract some part of lambda?

**Review by kugushev — APPROVED** [review]

**Review by pomkine — CHANGES_REQUESTED** [review]
Don't forget to fix new codeclimate issues (except imports). It is mainly about a length of lines (shoulb be < 100)

**Review by pomkine — APPROVED** [review]

**Pull request merged**